src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -0,0 +1,279 @@
+
+            GetRequest getRequest = new GetRequest(MASTER_KEY_INDEX).id(MASTER_KEY);
+            client.get(getRequest, ActionListener.wrap(getResponse -> {
+
+                if (!getResponse.isExists()) {
+
+                    // Generate new key and index
+                    final String masterKey = generateMasterKey();
+                    IndexRequest masterKeyIndexRequest = new IndexRequest(MASTER_KEY_INDEX).id(MASTER_KEY)
+                        .source(ImmutableMap.of(MASTER_KEY, masterKey))
+                        .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+                    client.index(masterKeyIndexRequest, ActionListener.wrap(indexResponse -> {
+                        // Set generated key to master
+                        logger.info("Master key has been initialized successfully");
+                        this.masterKey = masterKey;
+                        listener.onResponse(true);
+                    }, indexException -> {
+                        logger.error("Failed to index master key", indexException);
+                        listener.onFailure(indexException);
+                    }));
+
+                } else {
+                    // Set existing key to master
+                    logger.info("Master key has already been initialized");
+                    final String masterKey = (String) getResponse.getSourceAsMap().get(MASTER_KEY);
+                    this.masterKey = masterKey;
+                    listener.onResponse(true);
+                }
+            }, getRequestException -> {
+                logger.error("Failed to search for master key from master_key index", getRequestException);
+                listener.onFailure(getRequestException);
+            }));
+
+        } catch (Exception e) {
+            logger.error("Failed to retrieve master key from master_key index", e);
+            listener.onFailure(e);
+        }
+    }
+
+    /**
+     * Retrieves master key from system index if not yet set
+     */
+    protected void initializeMasterKeyIfAbsent() {
+        if (masterKey != null) {
+            return;
+        }
+
+        if (!clusterService.state().metadata().hasIndex(MASTER_KEY_INDEX)) {
+            throw new FlowFrameworkException("Master Key Index has not been initialized", RestStatus.INTERNAL_SERVER_ERROR);
+        } else {
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                GetRequest getRequest = new GetRequest(MASTER_KEY_INDEX).id(MASTER_KEY);
+                client.get(getRequest, ActionListener.wrap(response -> {
+                    if (response.isExists()) {
+                        this.masterKey = (String) response.getSourceAsMap().get(MASTER_KEY);
+                    } else {
+                        throw new FlowFrameworkException("Encryption key has not been initialized", RestStatus.NOT_FOUND);
```

Comment:
If we are keeping with master key when it has to do with the field we should change `encryption key` to `master key` here

Reply:
We should not mentioned anything about a key in logs. We can have some general message here.

Reply:
I have removed all mentions of master key in logs